### ISSUE TITLE
socket_last_error() equivalent API

### ISSUE THREAD

**Issue by dktapps** [krakjoe/pthreads#810]
Environment
==========

  * PHP: 7.2.1
  * pthreads: https://github.com/krakjoe/pthreads/commit/bc4aad0021c181df4a2f94301f1d5e800ddfe1f2
  * OS: windows x64

Summary
======
Equivalent API for the `socket_last_error()` function would be useful in code for detecting socket states.
Consider this code:

```php
$bindAddr = '0.0.0.0';
$bindPort = 19132;

$sock = socket_create(AF_INET, SOCK_DGRAM, SOL_UDP);
if(!socket_bind($sock, $bindAddr, $bindPort)){
	echo "[!] Can't bind to $bindAddr on port $bindPort, is something already using that port?" . PHP_EOL;
	exit(1);
}

socket_set_option($sock, SOL_SOCKET, SO_RCVTIMEO, ["sec" => 10, "usec" => 0]);

while(true){
	$status = @socket_recvfrom($sock, $buffer, 65535, 0, $source, $port);
	if($status !== false){
		//do things with the data
	}elseif(socket_last_error($sock) === SOCKET_ETIMEDOUT){
		//socket timeout
		break;
	}
}
```
cannot be ported to pthreads sockets because there is no way to determine what the last error from a socket was. I could parse the socket exception message if it threw one, but this is inconvenient.

(There are additional problems with porting this code such as setOption() not accepting arrays for options like timeout, but that is a different issue).

**Comment by dktapps** [comment]
This is resolved by the merge of #819 .

**Comment by dktapps** [comment]
After the effect, I realized I'm stupid - the error code is included in the thrown exception: https://github.com/krakjoe/pthreads/blob/master/src/socket.c#L71-L72 and could be retrieved with `getCode()`. Never mind - getLastError() is useful nonetheless...
